backend/routes/postsRoutes.js

```diff
@@ -0,0 +1,50 @@
+import { Router } from 'express'
+import { db } from '../lib/db'
+import { posts_db_name } from '../Utilities/API_utilities'
+
+
+const postsRouter = Router()
+
+postsRouter.get('/api/posts', (req, res) => {
+    db.getAll(posts_db_name).then(result => {
+        console.log('all posts', result, 'posts_db_name', posts_db_name)
+        return res.json(result)
+    }).catch(error => {
+        console.log('error getting all libraries', error)
+        return res.json(error)
+    })
+})
+
+postsRouter.post('/api/posts', function(req, res) {
+    db.insertOne(posts_db_name, req.body).then(response => {
+        res.status(201).json(response);
+    }).catch(error => {
+        res.status(406).json({'error': error})
```

Comment:
4XX errors are for client HTTP request errors (so if I passed in data for a bad request).

We probably want a 5XX error here - 500 would be the common one to use.

Reply:
agreed +1